api/src/infrastructure/docker.rs

```diff
@@ -490,6 +531,73 @@ impl DockerInfrastructure {
         Ok(())
     }
 
+    async fn get_existing_volumes(app_name: &String) -> Result<Vec<VolumeInfo>, ShipLiftError> {
```

Comment:
I'm a bit allergic to function names starting only with the verb get and I hate myself for using it when I'm unable to come up with a better name. Functions starting with get imply that it is something fast because it is a simple getter but here it is call to the Docker backend. What about `fetch_`?

```suggestion
    async fn fetch_existing_volumes(app_name: &String) -> Result<Vec<VolumeInfo>, ShipLiftError> {
```